oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -73,39 +73,52 @@ pub struct Runtime {
 }
 
 impl Runtime {
-    /// Configure and run the protobuf specified Application [`Configuration`]. After creating a
-    /// [`Runtime`], calling [`Runtime::stop`] will send termination signals to nodes and wait for
-    /// them to terminate. This returns a [`RuntimeRef`] reference to the created runtime, and a
-    /// writeable [`Handle`] to send messages into the runtime. To receive messages, creating a new
-    /// channel and passing the write [`Handle`] into the runtime will enable messages to be
-    /// read back out.
-    pub fn configure_and_run(config: Configuration) -> Result<(RuntimeRef, Handle), OakStatus> {
-        let runtime = Runtime {
-            configurations: config.nodes,
+    /// Creates a [`Runtime`] instance but does not start executing any node.
+    pub fn create(configuration: Configuration) -> Self {
+        Self {
+            configuration,
             terminating: AtomicBool::new(false),
             channels: channel::ChannelMapping::new(),
             nodes: Mutex::new(HashMap::new()),
             next_node_reference: AtomicU64::new(0),
-        };
+        }
+    }
 
-        let runtime = RuntimeRef(Arc::new(runtime));
+    /// Converts a [`Runtime`] instance into a [`RuntimeRef`].
+    pub fn into_ref(self) -> RuntimeRef {
+        RuntimeRef(Arc::new(self))
+    }
+
+    /// Configures and runs the protobuf specified Application [`Configuration`].
+    ///
+    /// After starting a [`Runtime`], calling [`Runtime::stop`] will send termination signals to
+    /// nodes and wait for them to terminate.
+    ///
+    /// Returns a [`RuntimeRef`] reference to the created runtime, and a writeable [`Handle`] to
```

Comment:
This PR splits the function to allow testing without starting the runtime. We need an Arc based return type for the nodes to have references to the runtime, `RuntimeRef` is just `Arc<Runtime>`. You could remove the wrapper type, but you would still have to use `Arc<Runtime>` as its the easiest way to pass a reference of the runtime into the node threads. Wrapping the `Arc<Runtime>` is really just a stylistic choice. When originally writing the code I thought it might be needed in more places and so would have more clear reason, although I'd lean towards using wrapper types either way.